pkg/queue/manager.go

```diff
@@ -489,6 +530,22 @@ func (m *Manager) DumpInadmissible() map[string]sets.Set[string] {
 	return dump
 }
 
+// Snapshot is a copy of the queues elements and inadmissible workloads list.
+func (m *Manager) Snapshot() []*kueue.Workload {
```

Comment:
I would prefer to avoid taking snapshot for all CQ in the same loop. In can be costly if we have many CQs, say 100. It was one of the concerns during the KEP review. 

The proposed solution is to have a fixed pool of workers that do the snapshots based on queue of snapshot tasks. Take a look here: https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility#in-memory-snapshot-of-the-clusterqueue. So, the idea is to have `takeSnapshot` function keying by the CQ name. Similarly as it is done for Job controller: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller.go#L233-L235. Then, we could have a function `enqueueSnapshotDelayed` to schedule taking snapshot after the configured interval.

We may sync up to discuss if there are some complications with this model.